frontend/src/util/bitwise-util.ts

```diff
@@ -6,15 +6,15 @@
 export const DAYS_IN_WEEK = 7;
 
 export const isBitSet = (bit: number, d: number, totalLen: number): boolean => (
-  (bit & (1 << (totalLen - d))) !== 0
+  (bit & (1 << (totalLen - 1 - d))) !== 0
```

Comment:
Could you write some explanation for these bitwise operations and then write some unit tests for it?

Reply:
Test cases have been added.